www/voll03/sp2/assets/js/main.js

```diff
@@ -0,0 +1,1138 @@
+
+            if (await getMovieDetails(response.data.results)) {
+                renderMovieListing(App.currentMovieList);
+                hideSpinner();
+
+                if (App.Pagination.maxPages > 1) {
+                    renderPagination(listSearched);
+                }
+            }
+        }).catch((error) => {
+            showError(error);
+            hideSpinner();
+        });
+    };
+
+    // helper function to set params for filtering
+    const getFilterParams = () => {
+        let params = "";
+        if (App.Filters.rating !== "") {
+            params += `&sort_by=${App.Filters.rating}&vote_count.gte=1000`;
+        }
+        if (App.Filters.genre !== "") {
+            params += `&with_genres=${App.Filters.genre}`;
+        }
+        if (App.Filters.yearFrom !== "") {
+            params += `&primary_release_date.gte=${App.Filters.yearFrom}-01-01`;
+        }
+        if (App.Filters.yearTo !== "") {
+            params += `&primary_release_date.lte=${App.Filters.yearTo}-12-31`;
+        }
+        if (App.Filters.country !== "") {
+            params += `&with_origin_country=${App.Filters.country}`;
+        }
+
+        return params;
+    };
+
+    // show filtered movies after setting filters and clicking oin "Apply filters" button
+    const listFiltered = () => {
+        refreshContentPanel("Filtered movies");
+        showSpinner();
+
+        const url = API.BASE + API.DISCOVER + API.KEY
+            + API.MISC_PARAMS + getFilterParams() + "&page=" + App.Pagination.currentPage;
+
+        axios.get(url).then(async (response) => {
+            if (response.data.total_results === 0) {
+                const alertMessageElement = $(`<p class="search-error">No results found for selected filters.</p>`);
+                DOM.contentPanel.append(alertMessageElement);
+                hideSpinner();
+                return;
+            }
+
+            App.currentMovieList = [];
+            App.currentListing = "Filtered movies";
+            App.Pagination.maxPages = response.data.total_pages;
+
+            if (await getMovieDetails(response.data.results)) {
+                renderMovieListing(App.currentMovieList);
+                hideSpinner();
+
+                if (App.Pagination.maxPages > 1) {
+                    renderPagination(listFiltered);
+                }
+            }
+        }).catch((error) => {
+            showError(error);
+            hideSpinner();
+        });
+    };
+
+    // show list of newly released movies
+    const listLatest = () => {
+        refreshContentPanel("Recently released");
+        DOM.pagination.remove();
+        showSpinner();
+
+        const url = API.BASE + API.NEWEST + API.KEY
+            + "&page=" + App.Pagination.currentPage;
+
+        axios.get(url).then(async (response) => {
+            App.currentMovieList = [];
+            App.currentListing = "Recently released";
+
+            if (await getMovieDetails(response.data.results)) {
```

Comment:
kontrola toho, zda se pozadavky dokoncily uspesne, nebo jeden z nich nedokoncilo uspesne, se dela pomoci `axios.all()` a `then` + `catch`, navic zavolani na movide detail byste mel udelat az v momente kdy se na jeden konkretni film klikne, protoze data z vyhledavani (pripadne `now_playing`) vam uz staci k tomu zobrazit dostatecne mnozstvi informaci o filmu pro uzivatele

Reply:
V datech z `now_playing` se u filmů nevyskytují informace o zemích původu (`production_countries`), které mám u jednotlivých filmů podle zadání uvádět, je zde pouze `original_language`. Proto tam mám call i na detaily k jednotlivým filmům. Chci se tedy zeptat - jestliže to budu zítra při obhajobě přepisovat, bude stačit uvést pouze jazyk?